libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -0,0 +1,136 @@
+
+    fun unregisterAll() {
+        observers.clear()
+        stop()
+    }
+
+    private fun stop() {
+        mapboxTimer.stopJobs()
+        jobControl.job.cancelChildren()
+        currentRequestId?.let { directionsSession.cancelRouteRequest(it) }
+    }
+
+    private fun requestRouteAlternatives() {
+        if (directionsSession.routes.isEmpty()) {
+            return
+        }
+
+        val routeOptionsResult = routeOptionsUpdater.update(
+            directionsSession.getPrimaryRouteOptions(),
+            tripSession.getRouteProgress(),
+            tripSession.getEnhancedLocation()
+        )
+
+        when (routeOptionsResult) {
+            is RouteOptionsUpdater.RouteOptionsResult.Success -> {
+                currentRequestId?.let { directionsSession.cancelRouteRequest(it) }
+                currentRequestId = directionsSession.requestRoutes(
+                    routeOptionsResult.routeOptions,
+                    routesRequestCallback
+                )
+            }
+            is RouteOptionsUpdater.RouteOptionsResult.Error -> {
+                logger.e(
+                    msg = Message("Route alternatives options are not available"),
+                    tr = routeOptionsResult.error
+                )
+            }
+        }
+    }
+
+    fun isDifferentRoute(directionsRoute: DirectionsRoute, callback: DifferentRouteCallback) {
+        jobControl.scope.launch {
+            val isDifferentRoute = navigator.isDifferentRoute(directionsRoute)
+            callback.result(isDifferentRoute)
+        }
+    }
+
+    private val routesRequestCallback = object : RoutesRequestCallback {
+        override fun onRoutesReady(routes: List<DirectionsRoute>) {
+            val routeProgress = tripSession.getRouteProgress()
+                ?: return
+            jobControl.scope.launch {
+                val alternatives = routes.filter { navigator.isDifferentRoute(it) }
+                val routes = mutableListOf<DirectionsRoute>()
+                routes.add(routeProgress.route)
+                routes.addAll(alternatives)
+                directionsSession.routes = routes
+                observers.forEach { it.onRouteAlternatives(routeProgress, alternatives) }
+            }
+        }
+
+        override fun onRoutesRequestFailure(throwable: Throwable, routeOptions: RouteOptions) {
+            logger.e(
+                msg = Message("Route alternatives request failed"),
+                tr = throwable
+            )
+        }
+
+        override fun onRoutesRequestCanceled(routeOptions: RouteOptions) {
+            logger.w(msg = Message("Route alternatives request canceled"))
+        }
+    }
+
+    override fun onNewMapMatcherResult(mapMatcherResult: MapMatcherResult) {
+        val routeProgress = tripSession.getRouteProgress()
+            ?: return
+        jobControl.scope.launch {
+            val alternatives = directionsSession.routes.filter { alternative ->
+                navigator.isAlternativeRoute(
```

Comment:
Instead of calls back and forth, it would be better to have @mapbox/navnative provide information about the route alternatives. Similar to how we're surfacing EHorizon information.